packages/payment-processor/src/payment/utils-near.ts

```diff
@@ -148,3 +164,92 @@ export const processNearPaymentWithConversion = async (
     throw new Error(`Could not pay Near request. Got ${e.message}`);
   }
 };
+
+export const processNearFungiblePayment = async (
+  walletConnection: WalletConnection,
+  network: CurrencyTypes.NearChainName,
+  amount: BigNumberish,
+  to: string,
+  paymentReference: string,
+  currencyAddress: string,
+  feeAddress: string,
+  feeAmount: BigNumberish,
+  callback: INearTransactionCallback | undefined = undefined,
+): Promise<void> => {
+  const fungibleContract = new Contract(walletConnection.account(), currencyAddress, {
+    changeMethods: ['ft_transfer_call'],
+    viewMethods: [],
+  }) as any;
+
+  const proxyAddress = erc20FeeProxyArtifact.getAddress(network, 'near');
+  await fungibleContract.ft_transfer_call({
+    args: {
+      receiver_id: proxyAddress,
+      amount,
+      msg: JSON.stringify({
+        fee_address: feeAddress,
+        fee_amount: feeAmount,
+        payment_reference: paymentReference,
+        to,
+      }),
+    },
+    gas: GAS_LIMIT_CONVERSION_TO_NATIVE,
+    amount: '1000000000000000000000000 '.toString(), // 1 yoctoNEAR deposit is mandatory for ft_transfer_call
+    ...callback,
+  });
+};
+
+type StorageBalance = {
+  total: string;
+  available: string;
+};
+
+// min. 0.00125 Ⓝ
+const MIN_STORAGE_FOR_FUNGIBLE = '1250000000000000000000';
+
+/**
+ * Stores the minimum deposit amount on the `paymentAddress` account for `tokenAddress`.
+ * This does not check the existing deposit, if any, and should be called if `isReceiverReady` is false.
+ * @param walletConnection
+ * @param tokenAddress
+ * @param paymentAddress
+ */
+export const storageDeposit = async (
```

Comment:
Not directly related to this PR, but if the payee does not have the required storage deposit, how do we plan to handle this on the FE.
Is it the payer who will have to perform an additional transaction to send the storage deposit amount to the payee's account, and then perform the actual payment ?

Reply:
So that's a key question I haven't answered yet. I should check how other dapps do. I think we should assume that the payer pays for this storage in this case (and we should thus increase the transaction deposit).

Reply:
@leoslr  cc @alexandre-abrioux